src/common/event_loop.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <queue>
+#include <mutex>
+
+#include <core/provider.pb.h>
+
+#include "types.h"
+
+namespace SailGame { namespace Common {
+
+using Core::ProviderMsg;
+
+class EventLoopSubscriber {
+public:
+    virtual void OnEventProcessed(const ProviderMsgPtr &) = 0;
+};
+
+class EventLoop {
+public:
+    static EventLoop &Create() {
+        static EventLoop eventLoop;
```

Comment:
static varible probably cause problem when you write unit test.
it won't be reset between two test cases.

Besides, what's the benefit? why you want to use static?

Reply:
that's true, I just considered using static variable is an easy way for singleton and durable storage.
changed to shared pointer now